ethcore/src/miner/miner.rs

```diff
@@ -482,6 +489,7 @@ impl Miner {
 				// Save proposal for later seal submission and broadcast it.
 				Seal::Proposal(seal) => {
 					trace!(target: "miner", "Received a Proposal seal.");
+					*self.next_mandatory_reseal.write() = Instant::now() + self.options.reseal_max_period;
```

Comment:
Then when mining Ethash chains, the `next_mandatory_reseal` will only be updated when you're actually successful in producing a solution? It's quite rare that you would do that every 2 minutes, so wouldn't miners end up periodically discarding non-stale candidates because of the max reseal period?